### ISSUE TITLE
Are there any converter available to transform StepF2 charts to SM5 counterpart?

### ISSUE THREAD

**Issue by ZM-J** [stepmania/stepmania#1931]
Your [wiki](https://github.com/stepmania/stepmania/wiki/Note-Types) introduces that,

> StepF2 uses a different note syntax, and these note types are not implemented in StepMania and they likely never will be, but they are documented here in case you ever need to convert a SF2 chart to SM5.

But are there any existed converter that can be used directly? Sure the converting process can be done, but it seems to be hard to do it easily :(

**Comment by RhythmLunatic** [comment]
StepMania doesn't support co-op x4, sudden, hidden, or the fake attribute for any note, so it wouldn't exactly be possible without converting them to regular notes. Which would break many charts. (Gargoyle Full co-op x2 is one chart that uses Fake+Sudden and hidden notes for special effects)

I've written a co-op to routine converter but it only works on charts without note attributes.

**Comment by ZM-J** [comment]
> StepMania doesn't support co-op x4, sudden, hidden, or the fake attribute for any note, so it wouldn't exactly be possible without converting them to regular notes. Which would break many charts. (Gargoyle Full co-op x2 is one chart that uses Fake+Sudden and hidden notes for special effects)
> 
> I've written a co-op to routine converter but it only works on charts without note attributes.

So you mean it's impossible to convert StepF2 charts with Sudden, Vanish or Hidden to equivalent SM5 charts? I'm mainly curious on this issue.

**Comment by ZM-J** [comment]
> StepMania doesn't support co-op x4, sudden, hidden, or the fake attribute for any note, so it wouldn't exactly be possible without converting them to regular notes. Which would break many charts.
> 
> I've written a co-op to routine converter but it only works on charts without note attributes.

So it has not been .
BTW, StepF2 seems not to be an open-source project, right?

**Comment by RhythmLunatic** [comment]
Yes, because StepMania doesn't have per-note modifiers. And fake notes aren't an attribute in SM5, just a tap note that doesn't count for score.

Something like this is not possible in SM5: https://youtu.be/-lLrRcyYxpo?t=101

**Comment by RhythmLunatic** [comment]
So my fork can parse StepF2 notes now and sudden/hidden/vanish and the fake note attribute works correctly. The code is honestly a mess because I tried getting it working as fast as possible, so I probably won't make a pull request until it's cleaned up. But you can compile my fork (and inspect my horrible code and weep) if you want to play StepF2 charts.
